subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/AbstractValidatingProperty.java

```diff
@@ -76,6 +79,10 @@ private static boolean isPresent(@Nullable Object value) {
         return value != null;
     }
 
+    private static boolean hasConfigurableValue(@Nullable Object value) {
```

Comment:
❓ To me it would look better to check the type of the value, not the value itself. We should know this, right? We should return `true` if the type extends `HasConfigurableValue` or it does _not_ extend `Provider`, right?

Reply:
Agreed offline to simplify the check to `value instanceof HasConfigurableValue`